20240627-BC-listapi.md

```diff
@@ -0,0 +1,159 @@
+# List API for the Dapr state component
+
+This proposal proposes implementing a List API in Dapr's state component. The List API will enable the retrieval of keys in a state store based on certain criteria, providing users with the necessary visibility into stored keys. List API results will **not** include the value. SDKs could provide the possibility to bulk get all the keys returned in a page.
+
+The requirements for the API are:
+
+- Ability to list all keys in a state store
+- Ability to list keys in a state store with a certain prefix
```

Comment:
not supported by all state stores

Reply:
Yes, @elena-kolevska has a table with her analysis and she will be sharing here. We can modify the proposal to reduce the feature set of the list API and maximize coverage across state stores but that will not be 100%. We already have that today for existing state store features. I don't think that evert single state store must support a feature to be added.

Reply:
Yes, I probably should have shared that from the beginning. I updated the proposal now with the table in an adendum.